src/cgutils.cpp

```diff
@@ -1023,7 +1023,7 @@ static void emit_typecheck(const jl_cgval_t &x, jl_value_t *type, const std::str
     //     return;
     // }
     Value *istype = emit_isa(x, type, &msg, ctx);
-    if (!isa<Constant>(istype)) {
+    if (!isa<ConstantInt>(istype)) {
```

Comment:
it assumes that `emit_isa` handled the return emitting the msg if necessary. We could return this as an explicit flag (`std::pair`) instead of assuming that the `builder` will only return certain types of objects.